kvstore/src/sstable.rs

```diff
@@ -92,8 +86,8 @@ impl SSTable {
         rows: &mut I,
         level: u8,
         max_table_size: u64,
-        data_wtr: &mut Writer,
-        index_wtr: &mut Writer,
+        data_wtr: &mut dyn Write,
```

Comment:
I'm curious, why all the trait objects (`dyn Write`) and not types that implement the Write trait (`W: Write`)?

Reply:
I was trying to avoid unnecessary generics to cut down on compile times, especially when this was still in the core core.

Probably unnecessary now that it's in a separate crate, but none of this should be a performance bottleneck. I figure later I can generify things if I need to.

Going forward I'll use generics unless there's a good reason not to.

Reply:
No worries, was just curious. Trait objects have their place too - just wasn't sure if this was one of them.